features/dsym_upload.feature

```diff
@@ -16,3 +16,18 @@ Feature: Uploading dSYMs to Bugsnag
         Then I should receive a request
         And the part "dsym" for request 0 is not null
         And the part "projectRoot" for request 0 equals "/Users/jenkins/build/app/"
+        And the payload field "apiKey" is null for request 0
+
+    Scenario: Uploading dSYMs with API key
+        When I upload dSYMS with options "--api-key 1234567890ABCDEF features/fixtures"
+        Then I should receive a request
+        And the part "dsym" for request 0 is not null
+        And the part "apiKey" for request 0 equals "1234567890ABCDEF"
+        And the payload field "projectRoot" is null for request 0
+
+    Scenario: Uploading dSYMs with a project root and API key
+        When I upload dSYMS with options "--project-root /Users/jenkins/build/app/ --api-key 1234567890ABCDEF features/fixtures"
+        Then I should receive a request
+        And the part "dsym" for request 0 is not null
+        And the part "projectRoot" for request 0 equals "/Users/jenkins/build/app/"
+        And the part "apiKey" for request 0 equals "1234567890ABCDEF"
```

Comment:
It would be good to have a scenario with conflicting api-key options and config-file options, to ensure config-file takes priority.

Reply:
I did see the test for this below, but I think it'd be good to have it in maze as well.

Reply:
I've added some more tests with a few variations. Need to clean up the test fastlane file as the plugin reacts differently depending on if variables are passed explicitly or read from the environment